File: PYTHON/ArrayAndStrings/Easy/mergeSortedArray.py
def mergeSortedArray(arr1,arr2):
    merged_array = []
    i,j = 0, 0

    while i< len(arr1) and j< len(arr2):
        if arr1[i] < arr2[j]:
            merged_array.append(arr1[i])
            i += 1
        else:
            merged_array.append(arr2[j])
            j += 1
    while i < len(arr1):
        merged_array.append(arr1[i])
        i += 1
    while j < len(arr2):
        merged_array.append(arr2[j])
        j += 1
    return merged_array

# Example usage:
if __name__ == "__main__":
    arr1 = [1, 3, 5, 7]
    arr2 = [2, 4, 6, 8]
    result = mergeSortedArray(arr1, arr2)
    print("Merged Sorted Array:", result)  # Output: Merged Sorted Array: [1, 2, 3, 4, 5, 6, 7, 8]

File: JAVASCRIPT/ArrayAndStrings/Easy/mergeSortedArray.js
// MERGE SORTED ARRAYS
function mergeSortedArray(arr1, arr2){
    let mergedArray = [];
    let i =0, j=0;

    while(i<arr1.length && j<arr2.length){
        if(arr1[i]<arr2[j]){
            mergedArray.push(arr1[i]);
            i++;
        }else{
            mergedArray.push(arr2[j]);
            j++;
        }
    }
    while(i<arr1.length){
        mergedArray.push(arr1[i]);
        i++;

    }
    while(j<arr2.length){
        mergedArray.push(arr2[j]);
        j++;
    }
    return mergedArray;
}


// Example usage:
const arr1 = [1, 3, 5];
const arr2 = [2, 4, 6];
console.log(mergeSortedArray(arr1, arr2)); // Output: [1, 2, 3, 4, 5, 6]